packages/datadog-instrumentations/src/next.js

```diff
@@ -141,6 +141,18 @@ function instrument (req, res, handler, error) {
   requests.add(req)
 
   const ctx = { req, res }
+  // Parse query parameters from request URL
+  if (queryParsedChannel.hasSubscribers && req.url) {
+    const url = new URL(req.url, 'http://n')
+    const query = {}
+    for (const key of url.searchParams.keys()) {
+      if (!query[key]) {
+        query[key] = url.searchParams.getAll(key)
+      }
+    }
```

Comment:
A better approach is to use the code `querystring` module, so you don't have to generate a complete URL object:

```suggestion
    const query = querystring.parse(req.url)
```

Reply:
```
> querystring.parse('/foo?bar=baz')
[Object: null prototype] { '/foo?bar': 'baz' }
```
With this approach without `new URL()` we need a regexp (or another step) to clean first param name

Reply:
Ah sorry, my bad. This would be the correct alternative:

```suggestion
    const { query } = url.parse(req.url, true)
```

But that's of course not 100% spec compliant (and now that I think of it, I guess the `querystring` module isn't either). However, I would still like to improve upon your implementation slightly:

```suggestion
    const { searchParams } = new URL(req.url, 'http://dummy')
    const query = {}
    for (const key of searchParams.keys()) {
      query[key] = searchParams.getAll(key)
    }
```

Reply:
why remove the `if (!query[key]) {` ? it was written like this in the previous instrumentation, probably for a reason no ?

Reply:
@simon-id Yeah that was a mistake on my part